dataproc/quickstart/quickstart.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This quickstart shows how you can use the Cloud Dataproc Client library to create a
+// Cloud Dataproc cluster, submit a PySpark job to the cluster, wait for the job to finish
+// and finally delete the cluster.
+
+// [START dataproc_quickstart]
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"time"
+
+	dataproc "cloud.google.com/go/dataproc/apiv1"
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+	dataprocpb "google.golang.org/genproto/googleapis/cloud/dataproc/v1"
+)
+
+// [END dataproc_quickstart]
+var (
+	projectID   string
+	region      string
+	clusterName string
+	jobFilePath string
+)
+
+func init() {
+	flag.StringVar(&projectID, "project_id", "", "Cloud Project ID, used for creating resources.")
+	flag.StringVar(&region, "region", "", "Region that resources should be created in.")
+	flag.StringVar(&clusterName, "cluster_name", "", "Name of Cloud Dataproc cluster to create.")
+	flag.StringVar(&jobFilePath, "job_file_path", "", "Path to job file in GCS.")
+}
+
+// [START dataproc_quickstart]
+func main() {
+	// TODO (Developer): Set the following variables.
+	// projectID = "your-project-id"
+	// region = "us-central1"
+	// clusterName = "your-cluster-name"
+	// jobFilePath = "gs://your_file/location"
+	// [END dataproc_quickstart]
+	flag.Parse()
```

Comment:
I think it's OK to show the flag parsing in the sample. It's what we do for other quickstarts. Maybe they don't all need to be flags? Could add comments saying to update as needed? The project ID needs to be a flag to test, though.

Reply:
For proper testing I believe these are all necessary